contracts/Pod.sol

```diff
@@ -47,76 +47,66 @@ contract Pod is
     |__________________________________*/
     IERC20Upgradeable public token;
     IERC20Upgradeable public ticket;
-    IERC20Upgradeable public pool;
+    IERC20Upgradeable public reward;
 
     // Initialized Contracts
     TokenFaucet public faucet;
```

Comment:
Let's add a function that allows the owner to set the faucet.  Just in case!  I'm sure we'll need to tweak it.

This also means we can remove `IERC20Upgradeable public reward;`, as it can change if the faucet is set.

Reply:
What do you mean by remove? What will we do with all the references to `reward` in the contract?

Just for now I added the function but required the new faucet.asset to == previous faucet.asset so the reward can't be changed... but you want `reward` to be updated too when TokenFaucet is updated? If so, what happens with unclaimed rewards from the previous TokenFaucet?

Are you imagining replacing `reward` with a call to `faucet.asset` in other functions like in `claimPodReward`?

Reply:
Hmm....this has opened up some interesting problems.  I think I spotted a sandwich attack.

Let's talk when you're up